src/common/load-plugins.js

```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+const resolve = require("resolve");
+
+function loadPlugins(options) {
+  options = Object.assign({ plugins: [] }, options);
+
+  const internalPlugins = [
+    require("../language-js"),
+    require("../language-css"),
+    require("../language-graphql"),
+    require("../language-markdown"),
+    require("../language-html")
+  ];
+
+  const externalPlugins = options.plugins.map(plugin => {
+    const pluginPath = resolve.sync(plugin, { basedir: process.cwd() });
+    return eval("require")(pluginPath);
```

Comment:
I wonder if there's any way this could support some sort of CLI-based API in addition to the JS ones, say some standard CLI commands that return information as JSON.

Doing so could allow writing Prettier plugins in languages other than JS, which would allow better integration for other language ecosystems.

Reply:
Hypothetically, yes. But it would require an RPC interface to call the various functions (`print`, `embed`, `parse`, etc)

Reply:
Not RPC per se since there's no remote ;)

Just like:

```
$ prettier-python parse <whatever>
```

Really this is all around https://github.com/prettier/prettier-python/issues/2.